### ISSUE TITLE
How to create a custom Authority/Entity with Form

### ISSUE THREAD

**Issue by giobber** [researchspace/researchspace#326]
As for the title

I installed latest ResearchSpace version by downloading the zip from https://researchspace.org/download/
I've tried to follow the various (sadly inconsistent) tutorial in the documentation but I'm not able to do basically anything outside the data already provided

I've tried to add a custom Authority/Entity following this tutorial https://documentation.researchspace.org/resource/rsp:RSTutorial. Till the 4th chapter I managed to make it work (althought from the images is obvious that the version used is a different one). I'm not sure about the 5th chapter, I've got no error, but my data seem to be different from the example.

The 6th chapter "Create a Form" is where the issue surface, if I try to copy the form from Architect (I don't have an authority called Person) to a new authority I called Actor, the new form doesn't work and raise a lot of error inside the webui

![Schermata_20240325_223150](https://github.com/researchspace/researchspace/assets/14294684/21696242-3092-4354-931c-e806c658af8e)

**Comment by doldman** [comment]
Can you share your Actor Form?

D

**Comment by giobber** [comment]
```xml
<semantic-form [[> rsp:FormDefaults]] 
               persistence='{"type": "sparql", "targetInsertGraphIri":"http:/researchspace.org/resource/g/data/architect"}'
               new-subject-template='/E21_Person/{{{{raw}}}}{{UUID}}{{{{/raw}}}}'
               fields='[[fieldDefinitions
                  classtype="http://www.w3.org/1999/02/22-rdf-syntax-ns#type"
                  has_type="http://www.cidoc-crm.org/cidoc-crm/P2_has_type"
                  
                  person_name="http://www.researchspace.org/pattern/person_name"
                  person_image="http://www.researchspace.org/ontology/PX_has_main_representation"
                  
                  gender="http://www.researchspace.org/person/PX_gender"
                  profession="http://www.researchspace.org/ontology/PX_profession"
                  nationality="http://www.researchspace.org/person/nationality"

                  person_birth="http://www.researchspace.org/pattern/person_birth"
                  person_death="http://www.researchspace.org/pattern/person_death"
                  
                  scopeNote="http://www.cidoc-crm.org/cidoc-crm/P3_has_note"
               ]]'
>

  <div class="form-scroll-area">
    <semantic-form-hidden-input for="classtype" default-value='http://www.cidoc-crm.org/cidoc-crm/E21_Person'> </semantic-form-hidden-input>
    <semantic-form-hidden-input for="has_type" default-value='http://www.researchspace.org/resource/vocab/system_entity/actor'> </semantic-form-hidden-input>
       
    <bs-tabs id="form-tabs" mount-on-enter=true>
      {{#if node}}
        <bs-tab event-key="details" title="Details">
          <mp-field-visualization subject='{{node}}'
                                  fields='[[
                                      fieldDefinitions
                                      has_type="http://www.cidoc-crm.org/cidoc-crm/P2_has_type"
                    
                                      person_name="http://www.researchspace.org/pattern/person_name"
                                      person_image="http://www.researchspace.org/ontology/PX_has_main_representation"
                                      
                                      gender="http://www.researchspace.org/person/PX_gender"
                                      profession="http://www.researchspace.org/ontology/PX_profession"
                                      nationality="http://www.researchspace.org/person/nationality"
                    
                                      person_birth="http://www.researchspace.org/pattern/person_birth"
                                      person_death="http://www.researchspace.org/pattern/person_death"
                                      
                                      scopeNote="http://www.cidoc-crm.org/cidoc-crm/P3_has_note"
                                    ]]'
      
                                  template='{{> template}}'>
    
            <template id="template">{{> rsp:ResourceFieldBasedVisualization subject="{{node}}" }} </template>
          </mp-field-visualization> 
        </bs-tab>

[[!-- <bs-tab event-key="rawData" title="Raw data">
         <mp-field-visualization subject='{{node}}'
                fields='[[fieldDefinitionsFromQuery (setQueryBindings "SELECT DISTINCT ?field WHERE {
                          ?resource ?field ?o . 
                          ?field a <http://www.researchspace.org/resource/system/fields/Field> 
                          } ORDER BY DESC(?field)" ) resource={{iri}} ]] {{/bind}}'
                template='{{> template}}'>
            <template id="template">{{> rsp:ResourceFieldBasedVisualization subject="{{node}}" }}</template>
          </mp-field-visualization>
        </bs-tab> --]]
      {{/if}}

      <bs-tab event-key="edit" title="Edit data">
        <div class="form-2column-container">
          <div>
            <semantic-form-text-input for="person_name" label="Person name" placeholder="Enter person name"></semantic-form-text-input>
            <semantic-form-select-input for="gender" label="Gender" placeholder="Select gender"> </semantic-form-select-input>
            <semantic-form-autocomplete-input for="profession" label="Profession" placeholder="Select profession"> </semantic-form-autocomplete-input>
            <semantic-form-autocomplete-input for="nationality" label="Nationality" placeholder="Select nationality"> </semantic-form-autocomplete-input>
        
            <semantic-form-composite-input for="person_birth" 
                                          label="Birth"
                                          new-subject-template="/E67_Birth"  
                                          fields='[[fieldDefinitions
                                                      classtype="http://www.w3.org/1999/02/22-rdf-syntax-ns#type"
                                                      birthTimeSpan="http://www.researchspace.org/pattern/person_birth/at_some_time_within"
                                                      birthPlace="http://www.researchspace.org/pattern/person_birth/place"
                                            ]]'>
        
              <semantic-form-hidden-input for="classtype" default-value='http://www.cidoc-crm.org/cidoc-crm/E67_Birth'> </semantic-form-hidden-input>
        
              <div style="display: flex; align-items: center; column-gap: 20px; margin-top: -15px;">
                <div style="flex: 1;">
                <semantic-form-datetime-input for='birthTimeSpan' label="Date" placeholder="Enter date of birth"></semantic-form-datetime-input>
                </div>
                <div style="flex: 1;">
                  <semantic-form-select-input for="birthPlace" label="Place" placeholder="Enter place of birth"></semantic-form-select-input>
                </div>
              
              </div>
            </semantic-form-composite-input>
        
            <semantic-form-composite-input for="person_death" 
                                          label="Death"
                                          new-subject-template="/E69_Death"  
                                          fields='[[fieldDefinitions
                                                      classtype="http://www.w3.org/1999/02/22-rdf-syntax-ns#type"
                                                      deathTimeSpan="http://www.researchspace.org/pattern/person_death/at_some_time_within"
                                                      deathPlace="http://www.researchspace.org/pattern/person_death/place"
                                            ]]'>
        
              <semantic-form-hidden-input for="classtype" default-value='http://www.cidoc-crm.org/cidoc-crm/E69_Death'> </semantic-form-hidden-input>
        
              <div style="display: flex; align-items: center; column-gap: 20px; margin-top: -15px;">
                <div style="flex: 1;">
                <semantic-form-datetime-input for='deathTimeSpan' label="Date" placeholder="Enter date of death"></semantic-form-datetime-input>
                </div>
                <div style="flex: 1;">
                  <semantic-form-select-input for="deathPlace" label="Place" placeholder="Enter place of death"></semantic-form-select-input>
                </div>
              
              </div>
            </semantic-form-composite-input>
        
            <semantic-form-text-input for="scopeNote" multiline=true label="Comment/Description" placeholder="Enter comment/description"> </semantic-form-text-input>

          </div>
          <div class="form-media-column">
            <div class="dragAndDrop-input-container">
              <semantic-form-drag-and-drop-input  for='person_image' label="Image" 
                                                  placeholder-item-template="<div class='placeholder-item'>
                                                                              <i class='fa fa-upload'></i>
                                                                              <div>Drag images or click to upload</div>
                                                                            </div>"
                                                  nested-form-template='{{{{raw}}}}{{> forms:image nested=true editable=true mode="new"}}{{{{/raw}}}}'>
              </semantic-form-drag-and-drop-input>
            </div>
            
          </div>

        </div>
      </bs-tab>

    </bs-tabs>

    <semantic-form-errors></semantic-form-errors>
  </div>
  
  [[> rsp:FormDefaultActions]] 
</semantic-form>
```

**Comment by doldman** [comment]
Please also go to settings - entity editor and choose your actor entity.

The page you are looking at is just the template code, not the interpreted
template.

D

**Comment by doldman** [comment]
I have tested your template and it works. You can now start to think about
different ways to present and edit the data. The architect default page is
one way (semantic table with links to a tab template with view and edit)
but there are other ways. For example,
rsp:SimpleCollectionBrowser?action=edit

BTW a new version of researchspace will be out soon.

Thanks,

D

**Comment by giobber** [comment]
I noticed that I haven't configured an Actor Authority outside of administration menu
I'll try to add it and check if now work, otherwise i'll send the other screenshot

Thanks for the help

Attached the screen for `settings > Authority and Entity Configuration > Actor`
![Schermata_20240325_234241](https://github.com/researchspace/researchspace/assets/14294684/dbaa3067-bed3-4bbd-88dd-f7282fbf99cd)

**Comment by doldman** [comment]
Ok. I used the same download and configured an authority resource with your
template and it worked fine.

Tell me how you get on.

D

**Comment by giobber** [comment]
Ok, I think i get it 

For start I was missing the Author as Authority in `ThinkingFrame > Resources > Authorities`
![Schermata_20240325_235911](https://github.com/researchspace/researchspace/assets/14294684/45662028-7cb9-4881-878b-903d2791619b)

As you pointed out , the template is used in other parts of the platform (as `entity editor` or in `Thinking Frames > Resources > Actor > edit`), I was missing the obvious.
![Schermata_20240326_000346](https://github.com/researchspace/researchspace/assets/14294684/dabf5395-199b-48c8-9c31-71bb1c52e4ff)

Now It work, I can view and edit data and customize the template

I'm not sure my configuration for Actor is exactly what I was expecting, the list of compatible object contains also other stuff see the next image

![Schermata_20240326_000044](https://github.com/researchspace/researchspace/assets/14294684/f0958718-bae4-4373-b076-35a142313d4b)

**Comment by doldman** [comment]
Hi Elia,

That's good news.

As I mentioned there is a new version of ResearchSpace which is being
prepared at the moment and will have more start-up configuration for new
users.

We are in parallel preparing documentation for this.

Essentially the authority configuration is designed to allow the system to
refer to entity URI's the correct form, or other tool.

There is a lot to ResearchSpace. If you need an orientation then I can
arrange an online meeting.

Glad you got this working.

Cheers,

Dominic

**Comment by denashams** [comment]
> Ok, I think i get it
> 
> For start I was missing the Author as Authority in `ThinkingFrame > Resources > Authorities` ![Schermata_20240325_235911](https://private-user-images.githubusercontent.com/14294684/316678230-45662028-7cb9-4881-878b-903d2791619b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzMzQxNTIsIm5iZiI6MTcxNjMzMzg1MiwicGF0aCI6Ii8xNDI5NDY4NC8zMTY2NzgyMzAtNDU2NjIwMjgtN2NiOS00ODgxLTg3OGItOTAzZDI3OTE2MTliLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIxVDIzMjQxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkODMwYTZhZDBjZmZmYTUxMWFiMDQ2MmYzMTAzNDA5NjYyZTQzYWUyMzQ1YmQyNWUzMGJmYTJjZTUzNzBmZTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ef_VBlbdepHpuwhowh6_AHZWqTuD4_oXDYexWfzmWWU)
> 
> As you pointed out , the template is used in other parts of the platform (as `entity editor` or in `Thinking Frames > Resources > Actor > edit`), I was missing the obvious. ![Schermata_20240326_000346](https://private-user-images.githubusercontent.com/14294684/316678263-dabf5395-199b-48c8-9c31-71bb1c52e4ff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzMzQxNTIsIm5iZiI6MTcxNjMzMzg1MiwicGF0aCI6Ii8xNDI5NDY4NC8zMTY2NzgyNjMtZGFiZjUzOTUtMTk5Yi00OGM4LTljMzEtNzFiYjFjNTJlNGZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIxVDIzMjQxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwOGYxNTIyMDczNmQ5NjBjZTg4OWFkOWNhNTc3MWMxMjNjNzgwY2Y4ZTI2YmUxZThmZTVmNTJmMjkwNGRhM2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YOvIyZmfjbrhMV8FkTBPR3XaBNYO8mmuFYCiYHsYwUo)
> 
> Now It work, I can view and edit data and customize the template
> 
> I'm not sure my configuration for Actor is exactly what I was expecting, the list of compatible object contains also other stuff see the next image
> 
> ![Schermata_20240326_000044](https://private-user-images.githubusercontent.com/14294684/316678445-f0958718-bae4-4373-b076-35a142313d4b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYzMzQxNTIsIm5iZiI6MTcxNjMzMzg1MiwicGF0aCI6Ii8xNDI5NDY4NC8zMTY2Nzg0NDUtZjA5NTg3MTgtYmFlNC00MzczLWIwNzYtMzVhMTQyMzEzZDRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIxVDIzMjQxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUxZWIzMzA4NGIwYzc3NGYxMDhiMGIyYjc1NGY1OGI1MWY0N2MzYThmMTNkNDhmYjY4ZDE1ODUxOTU5MWM3ZGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ocQKswPYknWePX-bQbT8mcIddH783qywOr1RA7tcJiU)



Hello,
I came across the same issue with creating new authorities and I was trying to use this solution. But my authority page seem to be missing the "New Authority" and "Edit" option.

![Screenshot 2024-05-22 012600](https://github.com/researchspace/researchspace/assets/18457333/51cca88f-93a8-4def-a331-e10037195665)

**Comment by giobber** [comment]
Hi, sorry for the late reply and really thank you for the support you give me

I'm trying in these days the new version, so far it works and seems a better looking. I found a new error but in the case I will open a new issue

We are working on an Open Linked Data project for an Italian association that deals with performative arts and historical memory. ResearchSpace is the most suitable tool we have found that meet our 
needs, and we hope to integrate it fully. If, as you wrote earlier, you are available for a orientation session, it could be quite useful for us.

Meanwhile I can close this issue
